### PR TITLE
Fix vgimage translate.

### PR DESCRIPTION
**Pull request by btracey**
The old code created scatter plots which were incorrectly offset on the canvas. This makes the correct offset.

Fixes Issue 179.

### CODE REVIEW

File: vg/vgimg/vgimg_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
..., 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0x28, 0xa2, 0x8a, 0x0, 0xa3, 0xac, 0x5e, 0xbe, 0x9f, 0xa5, 0x4f, 0x73, 0x12, 0xab, 0x48, 0xa0, 0x5, 0xdf, 0xf7, 0x41, 0x24, 0x0, 0x5b, 0xfd, 0x91, 0x9c, 0x9f, 0x60, 0x6a, 0xa4, 0x72, 0xdf, 0xe9, 0xfa, 0xad, 0xa5, 0xad, 0xd5, 0xea, 0xde, 0x47, 0x76, 0x1c, 0x73, 0x10, 0x46, 0x8d, 0x94, 0x6e, 0xc8, 0xc7, 0xf0, 0x76, 0xe7, 0x24, 0x12, 0xbc, 0x9c, 0xd6, 0xbc, 0x91, 0xa4, 0xd1, 0xbc, 0x72, 0x22, 0xbc, 0x6e, 0xa, 0xb2, 0xb0, 0xc8, 0x60, 0x7a, 0x82, 0x3d, 0x2a, 0x9d, 0x96, 0x8f, 0x61, 0xa7, 0x48, 0x64, 0xb5, 0xb7, 0x8, 0xe5, 0x76, 0x6e, 0x2c, 0x58, 0x85, 0xfe, 0xe8, 0x24, 0x9c, 0x2f, 0xb0, 0xe2, 0x80, 0x2f, 0x51, 0x45, 0x14, 0x0, 0x51, 0x45, 0x14, 0x0, 0x51, 0x45, 0x14, 0x0, 0x51, 0x45, 0x14, 0x0, 0x51, 0x45, 0x14, 0x1, 0xff, 0xd9}
```
Comment (kortschak) [inline vg/vgimg/vgimg_test.go]: Break this into 8 or 16 octet lines.

s/Answer/Want/g

File: vg/vgimg/vgimg_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	p.HideAxes()
+
+	c := vgimg.JpegCanvas{Canvas: vgimg.New(vg.Inch, vg.Inch)}
+	p.Draw(draw.New(c))
+	b := bytes.NewBuffer([]byte{})
+	n, err := c.WriteTo(b)
+	if err != nil {
+		t.Error(err)
+	}
+	if n != int64(len(issue179Answer)) {
```
Comment (kortschak) [inline vg/vgimg/vgimg_test.go]: This is redundant. bytes.Equal does this already and I'm not sure how valuable the differentiation is here.

**Comment by kortschak** [comment]
Would you change the commit comment to say that it was an error in all vgimg canvas plots, rather than making it scatter-specific.

LGTM after comments addressed.

File: vg/vgimg/vgimg_test.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		log.Fatal(err)
+	}
+	p, err := plot.New()
+	if err != nil {
+		log.Fatal(err)
+	}
+	p.Add(scatter)
+	p.HideAxes()
+
+	c := vgimg.JpegCanvas{Canvas: vgimg.New(vg.Inch, vg.Inch)}
```
Comment (kortschak) [inline vg/vgimg/vgimg_test.go]: Everytime I see a non-SI, non-ISO unit in scientific computing, I die a little inside.

**Comment by eaburns** [comment]
I'm nervous about starting the precedent of putting image data into byte arrays in test code. Can we make a testdata directory and load the image from a viewable image file instead? Then at least someone could open it in an image viewer.

**Comment by kortschak** [comment]
Yeah, I guess so. golang.org/x/image uses a testdata dir for test images.

**Comment by btracey** [comment]
PTAL

**Comment by btracey** [comment]
I'll fix the commit message when I rebase.

**Comment by kortschak** [comment]
LGTM

File: vg/vgimg/vgimg_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		log.Fatal(err)
+	}
+	p, err := plot.New()
+	if err != nil {
+		log.Fatal(err)
+	}
+	p.Add(scatter)
+	p.HideAxes()
+
+	c := vgimg.JpegCanvas{Canvas: vgimg.New(5.08*vg.Centimeter, 5.08*vg.Centimeter)}
```
Comment (kortschak) [inline vg/vgimg/vgimg_test.go]: Ha ha!

Comment (btracey) [inline vg/vgimg/vgimg_test.go]: I'm not sure which is more sad. The number of times I was given numbers like this in undergrad engineering classes, or the number of times I had to use the alternating units of pound-mass and slugs.

Comment (kortschak) [inline vg/vgimg/vgimg_test.go]: One day the US will join us in the century of the fruit bat.

Comment (eaburns) [inline vg/vgimg/vgimg_test.go]: :)
I think that we should just regenerate the test data to use centimeters.

**Comment by eaburns** [comment]
LGTM

File: vg/vgimg/vgimg_test.go (hunk 4 above)
Comment (kortschak) [inline vg/vgimg/vgimg_test.go]: That's the joke.

**Pull request merged**